src/dialogs/interfaces/abstractDialogFieldClass.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * This is abstract controller for DialogField components
+ * @memberof miqStaticAssets.dialog
+ * @ngdoc controller
+ * @name DialogFieldClass
+ */
+export abstract class DialogFieldClass {
+
+  public field: any;
+  public onUpdate: any;
+  public options: any;
+  public inputDisabled: boolean;
+    /*@ngInject*/
+  constructor() {
```

Comment:
This is was needed in order to call a constructor in the class this extended.  Otherwise typescript was giving errors.